### PR TITLE
Update Azure SDK to v53.4.0

### PR DESCRIPTION
**Pull request by magodo**
This SDK version is needed to implement #10960.

### CODE REVIEW

File: azurerm/internal/services/sentinel/sentinel_data_connector_aws_cloud_trail.go [hunk 1]
```diff
@@ -96,7 +96,7 @@ func resourceSentinelDataConnectorAwsCloudTrailCreateUpdate(d *schema.ResourceDa
 				},
 			},
 		},
-		Kind: securityinsight.KindAmazonWebServicesCloudTrail,
+		Kind: securityinsight.KindBasicDataConnectorKindAmazonWebServicesCloudTrail,
```
Comment (tombuildsstuff) [inline azurerm/internal/services/sentinel/sentinel_data_connector_aws_cloud_trail.go:99]: this is a breaking change in a minor version of the SDK?

Comment (magodo) [reply azurerm/internal/services/sentinel/sentinel_data_connector_aws_cloud_trail.go:99]: @ArcturusZhang Would you please clarify?

Comment (ArcturusZhang) [reply azurerm/internal/services/sentinel/sentinel_data_connector_aws_cloud_trail.go:99]: We currently do not block the breaking changes in preview packages in minor versions

**Review by tombuildsstuff — APPROVED** [review]
LGTM 👍

**Pull request merged**